crates/polkavm-common/src/writer.rs

```
@@ -34,6 +34,19 @@ impl From<(u32, Instruction)> for InstructionBuffer {
     }
 }
 
+impl From<Vec<u8>> for InstructionBuffer {
+    fn from(value: Vec<u8>) -> Self {
+        let mut bytes = [0u8; program::MAX_INSTRUCTION_LENGTH];
+        let slice_size = value.len().min(program::MAX_INSTRUCTION_LENGTH);
```

Shouldn't silently truncate. Add a `TryFrom` impl instead.

---

I removed this, as it is not needed anymore. `InstructionBuffer` is used directly inside `InstructionOrBytes`.